xonsh/parsers/base.py

```diff
@@ -19,13 +22,55 @@
 from xonsh.tokenize import SearchPath, StringPrefix
 from xonsh.lazyasd import LazyObject
 from xonsh.parsers.context_check import check_contexts
+from xonsh.events import events
+from xonsh.tools import print_exception
```

Comment:
Maybe put it in a new module then in the parser subpackage? I still think that is a little weird. The parser doesn't actually use this event though. Could you clarify how this helps consistency?  I had put this in base_shell originally because that is what gets called by all of the frontends.